cmd/meradm/server.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"errors"
+
+	"strconv"
+
+	"fmt"
+
+	"strings"
+
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"github.com/sky-uk/merlin/types"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+var serverCmd = &cobra.Command{
+	Use:   "server [add|edit|del]",
+	Short: "Modify a real server",
+}
+
+func validServiceIDIPPort(_ *cobra.Command, args []string) error {
+	if len(args) != 2 {
+		return errors.New("requires two arguments")
+	}
+	b := []byte(args[1])
+	if !ipPortRegex.Match(b) {
+		return errors.New("must be ip:port")
+	}
+	return nil
+}
+
+var addServerCmd = &cobra.Command{
+	Use:   "add [serviceID] [ip:port]",
+	Short: "Add a real server",
+	Args:  validServiceIDIPPort,
+	RunE:  addServer,
+}
+
+var editServerCmd = &cobra.Command{
+	Use:   "edit [serviceID] [ip:port]",
+	Short: "Edit a real server",
+	Args:  validServiceIDIPPort,
+	RunE:  editServer,
+}
+
+var deleteServerCmd = &cobra.Command{
+	Use:   "del [serviceID] [ip:port]",
+	Short: "Delete a real server",
+	Args:  validServiceIDIPPort,
+	RunE:  deleteServer,
+}
+
+var (
+	weight  string
+	forward string
+)
+
+func init() {
+	rootCmd.AddCommand(serverCmd)
+	serverCmd.AddCommand(addServerCmd)
+	serverCmd.AddCommand(editServerCmd)
+	serverCmd.AddCommand(deleteServerCmd)
+
+	for _, f := range []*pflag.FlagSet{addServerCmd.Flags(), editServerCmd.Flags()} {
+		f.StringVarP(&weight, "weight", "w", "", "weight of real server, used by the IPVS scheduler")
+		f.StringVarP(&forward, "forward", "f", "", "forwarding method, one of [route|tunnel|masq]")
+	}
+
+	addServerCmd.MarkFlagRequired("weight")
+	addServerCmd.MarkFlagRequired("forward")
+}
+
+func initServer(serviceID string, hostPort string) (*types.RealServer, error) {
```

Comment:
You missed `hostPort` and `host` in this function.

Reply:
Okay, done.